src/field_theory/finite.lean

```diff
@@ -1,138 +1,247 @@
 /-
 Copyright (c) 2018 Chris Hughes. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Chris Hughes
+Authors: Chris Hughes, Joey van Langen, Casper Putz
 -/
 import group_theory.order_of_element
 import data.polynomial
 import data.equiv.ring
 import data.zmod.basic
+import linear_algebra.basis
+import algebra.geom_sum
 
-universes u v
-variables {α : Type u} {β : Type v}
+/-!
+# Finite fields
 
-open function finset polynomial nat
+This file contains basic results about finite fields.
+Throughout most of this file, K denotes a finite field with q elements.
+
+## Main results
+
+1. Every finite integral domain is a field (`field_of_integral_domain`).
+2. The unit group of a finite field is a cyclic group of order q - 1.
+   (`finite_field.is_cyclic` and `card_units`)
+3. `sum_pow_units`: The sum of x^i, where x ranges over the units of K, is
+   | q-1 if q-1 ∣ i
+   | 0   otherwise
```

Comment:
```suggestion
   - `q-1` if `q-1 ∣ i`
   - `0`   otherwise
```
I think markdown will put all of this text on the same line unless we do something like this.